packages/@ama-sdk/schematics/schematics/code-generator/open-api-cli-generator/open-api-cli.generator.ts

```diff
@@ -1,14 +1,25 @@
 import {CodeGenerator} from '../code-generator';
 import {spawn, SpawnOptions} from 'node:child_process';
 import {defaultTypescriptGeneratorOptions, OpenApiCliOptions} from './open-api-cli.options';
+import * as path from 'node:path';
 
 /**
  * Manage the schematic to generate a sdk using the @openapitools/openapi-generator-cli
  */
 export class OpenApiCliGenerator extends CodeGenerator<OpenApiCliOptions> {
-  private packageManager = process.env && process.env.npm_execpath && process.env.npm_execpath.indexOf('yarn') === -1 ? 'npm' : 'yarn';
+
+  protected readonly packageManager: 'npm' | 'yarn';
+  protected get packageManagerRunner(): 'npx' | 'yarn' {
+    return this.packageManager === 'npm' ? 'npx' : 'yarn';
+  }
+
   /** @inheritDoc */
-  protected generatorName = 'openapi-codegen-npm-cli';
+  protected readonly generatorName = 'openapi-codegen-npm-cli';
+
+  constructor(options?: { packageManager?: 'npm' | 'yarn' | '' }) {
+    super();
+    this.packageManager = options?.packageManager || (process.env?.npm_execpath?.includes('yarn') ? 'yarn' : 'npm');
```

Comment:
can't we do a global helper for that ?
we have this kind of piece of code everywhere